lib/openstax/aws/s3_text_file.rb

```diff
@@ -18,11 +18,20 @@ def read
       object.get.body.read
     end
 
-    def write(contents, content_type='text/plain')
-      object.put(
-        body: StringIO.new(contents),
-        content_type: content_type
-      )
+    def get
+      object.load
+      object.get
+    end
+
+    def write(string_contents:, content_type:'text/plain', cache_control: nil)
+      args = {
+        body: StringIO.new(string_contents)
+      }
+
+      args[:content_type] = content_type if !content_type.nil?
```

Comment:
`args[:content_type] = content_type if content_type` ?  (a bit more concise)

Reply:
Yeah... for some reason I generally don't do that.  Some sort of long-lasting objection to boolean checks on non boolean things in C++ I think :-)